### PR TITLE
Er/ct 303 004 simple snapshot

### LINKED ISSUE #4806: [CT-303] 004_simple_snapshot_tests

### PR DESCRIPTION
**Pull request by emmyoop**
resolves #4806

### Description

WIP 004 test conversion

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] ~~I have updated the `CHANGELOG.md` and added information about my change~~

### CODE REVIEW

File: tests/functional/simple_snapshot/test_basic_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+@pytest.mark.usefixtures("project")
+class TestBasicCustomNamespace(CustomNamespace):
+    def test_basic_snapshot(project, basic_snapshot):
+        basic_snapshot
+
+
+@pytest.mark.usefixtures("project")
+class TestRefCustomNamespace(CustomNamespace):
+    def test_basic_ref(project, basic_ref):
```
Comment (ChenyuLInx) [inline tests/functional/simple_snapshot/test_basic_snapshot.py:217]: If the mark.usefixtures("project") is stated for the whole calss, is the project fixture here still needed?

File: tests/functional/simple_snapshot/test_slow_query_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+def snapshots():
+    return {"snapshot.sql": snapshots_slow__snapshot_sql}
+
+
+@pytest.fixture(scope="class")
+def tests():
+    return {"test_timestamps.sql": test_snapshots_slow__test_timestamps_sql}
+
+
+def test_slow(project):
```
Comment (ChenyuLInx) [inline tests/functional/simple_snapshot/test_slow_query_snapshot.py:74]: Do we still need this one?

**Comment by gshank** [comment]
Fixtures shouldn't be used for executing tests. They are intended to be used only for setting up the data/project necessary for executing tests. I'm going to take a deeper look and see if I can come up with some suggestions for how to do it.

**Comment by gshank** [comment]
A bunch of the tests fail when doing `pytest -v tests/functional/simple_snapshot`

**Review by gshank — CHANGES_REQUESTED** [review]

**Review by gshank — APPROVED** [review]
Looks good!

**Pull request merged**